flink-ml-lib/src/main/java/org/apache/flink/ml/feature/vectorassembler/VectorAssembler.java

```diff
@@ -0,0 +1,187 @@
+
+import org.apache.commons.lang3.ArrayUtils;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * A feature transformer that combines a given list of input columns into a vector column. Types of
+ * input columns must be either vector or numerical value.
+ */
+public class VectorAssembler
+        implements Transformer<VectorAssembler>, VectorAssemblerParams<VectorAssembler> {
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+    private static final double RATIO = 1.5;
+
+    public VectorAssembler() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public Table[] transform(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) inputs[0]).getTableEnvironment();
+        RowTypeInfo inputTypeInfo = TableUtils.getRowTypeInfo(inputs[0].getResolvedSchema());
+        RowTypeInfo outputTypeInfo =
+                new RowTypeInfo(
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(), TypeInformation.of(Vector.class)),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getOutputCol()));
+        DataStream<Row> output =
+                tEnv.toDataStream(inputs[0])
+                        .flatMap(
+                                new AssemblerFunc(getInputCols(), getHandleInvalid()),
+                                outputTypeInfo);
+        Table outputTable = tEnv.fromDataStream(output);
+        return new Table[] {outputTable};
+    }
+
+    private static class AssemblerFunc implements FlatMapFunction<Row, Row> {
+        private final String[] inputCols;
+        private final String handleInvalid;
+
+        public AssemblerFunc(String[] inputCols, String handleInvalid) {
+            this.inputCols = inputCols;
+            this.handleInvalid = handleInvalid;
+        }
+
+        @Override
+        public void flatMap(Row value, Collector<Row> out) throws Exception {
+            try {
+                Object[] objects = new Object[inputCols.length];
+                for (int i = 0; i < objects.length; ++i) {
+                    objects[i] = value.getField(inputCols[i]);
+                }
+                Vector assembledVector = assemble(objects);
+                out.collect(Row.join(value, Row.of(assembledVector)));
+            } catch (Exception e) {
+                switch (handleInvalid) {
+                    case VectorAssemblerParams.ERROR_INVALID:
+                        throw e;
+                    case VectorAssemblerParams.SKIP_INVALID:
+                        return;
+                    case VectorAssemblerParams.KEEP_INVALID:
+                        /*
+                         * If KEEP_INVALID is set, all input row values will be reserved and a null
```

Comment:
Would it be simpler to remove this command? Or we can copy/paste the Java doc for "keep" from `VectorAssemblerParams`.

Reply:
OK